BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```diff
@@ -404,6 +404,48 @@ - (BOOL)deleteCachedInfoForUserId:(NSString *)userId associateId:(NSString *)ass
     return success;
 }
 
+- (NSDictionary *)associateIdToBackgroundSessionIdAndSessionTaskIdsForUserId:(NSString *)userId error:(NSError **)outError
+{
+    if (userId == nil) {
+        return nil;
+    }
+
+    NSError *error = nil;
+    NSArray *associateIds = [self associateIdsForUserId:userId error:&error];
+    NSMutableDictionary *associateIdToBackgroundSessionIdAndSessionTaskIds = [NSMutableDictionary new];
+
+    //iterate through users/$userId/$associateId subdirs to get its backgroundSessionId and sessionTaskId
+    for (NSString *associateId in associateIds) {
+
+        BOXURLBackgroundSessionIdAndSessionTaskId *backgroundSessionIdAndSessionTaskId = [self backgroundSessionIdAndSessionTaskIdForUserId:userId associateId:associateId error:&error];
+
+        if (backgroundSessionIdAndSessionTaskId != nil) {
+            associateIdToBackgroundSessionIdAndSessionTaskIds[associateId] = backgroundSessionIdAndSessionTaskId;
+        } else if (error == nil) {
```

Comment:
don't think there are other errors that we need to clean up. Let me know if you do.